api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -788,7 +788,8 @@ message Cluster {
 
   // Optional maximum requests for a single upstream connection. This parameter
   // is respected by both the HTTP/1.1 and HTTP/2 connection pool
-  // implementations. If not specified, there is no limit. Setting this
+  // implementations. If not specified, there is no limit for downstream connections.
```

Comment:
The default limit for h2 upstream connections is to prevent the exhaustion of stream id.

But the max_requests_per_connection of donwstream is used to limit the streams which server could receive and it is not related to stream id exhaustion of server, actualy it would be handled automatically by underlying nghttp2. 

So I think it is ok to stay as it is. WDTY? @yanavlasov @htuch

Reply:
What is the nghttp2 behavior in this situation? Whether it's done by Envoy core code or nghttp2, the end user wants to know what the actual behavior is. It can't be unlimited due to the fundamentals of HTTP/2.

Reply:
The processing of `max_requests_per_connection` is implemented by envoy core code.

For upstream it help avoid the stream id exhaustion, because the client would call underlying nghttp2 to create stream but only under the limit of `max_requests_per_connection`.

But for donwstream,  `max_requests_per_connection` only limit the streams that server could receive (ref: https://github.com/envoyproxy/envoy/blob/main/source/common/http/conn_manager_impl.cc#L295) which is not related to the stream id exhaustion of server, because the stream id of server will only be consumed when the server call the underlying nghttp2 to create new stream. 

Maybe the advice of @alyssawilk would be helpful :)

Reply:
Yes, but the point is that the downstream connection has an implicit upper bound. We should explain this, since it seems to be a source of confusion (see https://github.com/envoyproxy/envoy/issues/19219).